src/vs/code/electron-main/app.ts

```diff
@@ -1216,6 +1217,11 @@ export class CodeApplication extends Disposable {
 
 		// Crash reporter
 		this.updateCrashReporterEnablement();
+
+		if (isMacintosh && isRosettaTranslated() === ProcTranslated.Translated) {
```

Comment:
You can use [app.runningUnderARM64Translation](https://github.com/electron/electron/blob/main/docs/api/app.md#apprunningunderarm64translation-readonly-macos-windows) which is available from the runtime instead.

Reply:
i didn't know that, i'm committing a fix